src/components/RegisterPage/MainRegister.tsx

```diff
@@ -1,50 +1,67 @@
 import React, { useState, useRef, useEffect } from "react";
+import { UseQueryResult } from "react-query";
 import { useQueryParam } from "use-query-params";
 
+import { useParams } from "react-router-dom";
+
 import MAIN from "../main_component";
-import HEADER from "../header_main";
+import { Header } from "./header";
 import SELECT_MULTI from "../select_multi";
 import SELECT_SINGLE from "../select_single";
-import TU_LIST from "../tu_list";
 
 import config, {
   mainQueryParamsConfig,
   maxYear,
   minYear,
   defaultYear,
 } from "../../app_config";
-import { nest_tu_names } from "../../data/filter_year_unit";
-import { useResizeObserver } from "../../helpers/hooks";
-import { filter_year_unit } from "../../data/filter_year_unit";
-import { mathClamp, validateTreatmentUnits } from "../../helpers/functions";
 
-import { StatisticData, Description, TreatmentUnit, AggData } from ".";
+import { useResizeObserver, useUnitNamesQuery } from "../../helpers/hooks";
+import { mathClamp, validateTreatmentUnits } from "../../helpers/functions";
+import { RegisterNames } from ".";
+import { UnitNameList } from "./unitnamelist";
 
-const { med_field, app_text } = config;
+const { app_text } = config;
 
 interface MainRegisterProps {
-  tu_names: TreatmentUnit[];
-  sortedIndicatorHospital: StatisticData[];
-  sortedIndicatorHf: StatisticData[];
-  sortedIndicatorRhf: StatisticData[];
-  sortedIndicatorNation: StatisticData[];
-  description: Description[];
-  isLoading: boolean;
+  registerNames: RegisterNames[];
 }
 
-const MainRegister: React.FC<MainRegisterProps> = ({
-  isLoading,
-  tu_names,
-  sortedIndicatorHospital,
-  sortedIndicatorHf,
-  sortedIndicatorRhf,
-  sortedIndicatorNation,
-  description,
-}) => {
+const MainRegister: React.FC<MainRegisterProps> = ({ registerNames }) => {
+  const { tab }: { tab: string } = useParams();
+  const tabNames: { label: string; value: string }[] = [
+    { value: "sykehus", label: "Sykehus" },
+    { value: "boomraade", label: "Boområde" },
+    { value: "datakvalitet", label: "Datakvalitet" },
+  ];
+  const context =
+    tab === "sykehus"
+      ? "caregiver"
+      : tab === "boomraade"
+      ? "recident"
+      : tab === "datakvalitet"
+      ? "coverage"
+      : "caregiver";
+  const activeTab = tab;
+  const [nestedUnitNames, updateNestedUnitNames] = useState<[]>([]);
+
+  const [optstu, updateOptsTU] = useState<[]>([]);
+  const unitNamesQuery: UseQueryResult<any, unknown> = useUnitNamesQuery(
+    "all",
+    context
+  );
+  useEffect(() => {
+    if (unitNamesQuery.isSuccess) {
+      updateNestedUnitNames(unitNamesQuery.data.nestedUnitNames);
+      updateOptsTU(unitNamesQuery.data.opts_tu);
+    }
+  }, [unitNamesQuery]);
```

Comment:
Du kan vel bare bruke resultatet direkte og trenger ikke effect og state?

```suggestion
  const unitNamesQuery: UseQueryResult<any, unknown> = useUnitNamesQuery(
    "all",
    context
  );
  
  const nestedUnitNames = unitNamesQuery.data?.nestedUnitNames ?? [];
  const optstu = unitNamesQuery.data?.opts_tu ?? [];
```